public/app/features/trails/ActionTabs/MetricOverviewScene.tsx

```diff
@@ -0,0 +1,171 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import {
+  QueryVariable,
+  SceneComponentProps,
+  SceneFlexItem,
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectState,
+  SceneVariableSet,
+  VariableDependencyConfig,
+} from '@grafana/scenes';
+import { Stack, useStyles2, TextLink } from '@grafana/ui';
+
+import PrometheusLanguageProvider from '../../../plugins/datasource/prometheus/language_provider';
+import { PromMetricsMetadataItem } from '../../../plugins/datasource/prometheus/types';
+import { getDatasourceSrv } from '../../plugins/datasource_srv';
+import { ALL_VARIABLE_VALUE } from '../../variables/constants';
+import { trailDS, VAR_DATASOURCE_EXPR, VAR_GROUP_BY, VAR_METRIC_EXPR } from '../shared';
+import { getMetricSceneFor } from '../utils';
+
+import { getLabelOptions } from './utils';
+
+export interface MetricOverviewSceneState extends SceneObjectState {
+  metadata?: PromMetricsMetadataItem;
+  loading?: boolean;
+}
+
+export class MetricOverviewScene extends SceneObjectBase<MetricOverviewSceneState> {
+  protected _variableDependency = new VariableDependencyConfig(this, {
+    variableNames: [VAR_DATASOURCE_EXPR],
+    onReferencedVariableValueChanged: this.onReferencedVariableValueChanged.bind(this)
+  });
+
+  constructor(state: Partial<MetricOverviewSceneState>) {
+    super({
+      $variables: state.$variables ?? getVariableSet(),
+      ...state,
+    });
+
+    this.addActivationHandler(this._onActivate.bind(this));
+  }
+
+  private getVariable(): QueryVariable {
+    const variable = sceneGraph.lookupVariable(VAR_GROUP_BY, this)!;
+    if (!(variable instanceof QueryVariable)) {
+      throw new Error('Group by variable not found');
+    }
+
+    return variable;
+  }
+
+  private _onActivate() {
+    this.updateMetadata();
+
+    const metricScene = getMetricSceneFor(this);
+    metricScene.subscribeToState((newState, oldState) => {
+      if (newState.metric !== oldState.metric) {
+        this.updateMetadata();
+      }
+    });
```

Comment:
I don't think this is needed, I could not get line 61 to trigger. When a metric is changed we build a new MetricScene (which builds a new MetricOverview),